specs/agents/tracing-spans-dropped-stats.md

```diff
@@ -0,0 +1,57 @@
+[Agent spec home](README.md) > [Handling huge traces](tracing-spans-handling-huge-traces.md) > [Collecting statistics about dropped spans](tracing-spans-dropped-stats.md)
+
+## Collecting statistics about dropped spans
+
+To still retain some information about dropped spans (for example due to [`transaction_max_spans`](tracing-spans-limit.md) or [`exit_span_min_duration`](tracing-spans-drop-fast-exit.md)),
+agents SHOULD collect statistics on the corresponding transaction about dropped spans.
+These statistics MUST only be sent for sampled transactions.
+
+### Use cases
+
+This allows APM Server to consider these metrics for the service destination metrics.
+In practice,
+this means that the service map, the dependencies table,
+and the backend details view can show accurate throughput statistics for backends like Redis,
+even if most of the spans are dropped.
+
+This also allows the transaction details view (aka. waterfall) to show a summary of the dropped spans.
+
+### Data model
+
+This is an example of the statistics that are added to the `transaction` events sent via the intake v2 protocol.
+
+```json
+{
+  "dropped_spans_stats": [
+    {
+      "type": "external",
+      "subtype": "http",
+      "destination_service_resource": "example.com:443",
+      "outcome": "failure",
```

Comment:
We don't drop spans with `outcome: failure` right?

Reply:
We don't compress them, but spans with `outcome: failure` can still be dropped because of `transaction_max_spans` limit. In that case we need to collect statistics for those spans as well.